tests/cli/test_cli_parser.py

```diff
@@ -268,3 +269,13 @@ def test_invalid_choice_raises_for_export_format_in_db_export_archived_command(
             f"--export-format: invalid choice: '{export_format}' "
             "(choose from 'csv'), see help above.\n"
         )
+
+    def test_cli_run_time(self):
+        setup_code = "import subprocess"
+        timing_code = 'subprocess.run(["airflow", "--help"])'
+        # Limit the number of samples otherwise the test will take a very long time
+        num_samples = 3
+        threshold = 3.5
+        timing_result = timeit.timeit(stmt=timing_code, number=num_samples, setup=setup_code) / num_samples
+        # Average run time of Airflow CLI should at least be within 3.5s
+        assert timing_result < threshold
```

Comment:
Each `airflow --help` currently takes more than three seconds? That’s bad.

Reply:
Each command must complete in `<` less than 3.5 seconds. On my machine it usually takes are 2-2.5 seconds in rare cases spiking up to 3 if you CPU is under load for something, etc.

But yeah, the CLI module lives in the same code base as everything else, so to run it all of Airflow must be loaded (including the heavy hitters of `conf`/`settings`, etc)

Reply:
We should have different timings for different commands IMHO. And I agree with @uranusjr that if basic commands take > 3 seconds, this is wrong. Some of them ("config" that I worked on recently in https://github.com/apache/airflow/pull/32604 need to initialize providers now (and I deliberately introduced lazy-loading the providers-manager for config in my change  to only run it when actually needed. But maybe this is an indication that we still need to improve some of that . @uranusjr had done a lot of those lazy-loading things to make the time smaller for CLIs and some initialization, but 3 seconds for "most" commands seems really off.

Reply:
I agree with all of this, but we can also agree that it is out of scope for this PR, right? I just want to establish some kind of threshold now in the most basic sense (i.e. just a simple `--help`) and we can have much more follow up work to optimize it and lower the threshold once we make more optimizations.

Reply:
A different threshold for each command sounds great but also sounds like a lot of work and way more chances to break something for what feels like marginal benefit to me.  Either way, I'd propose we get the test implemented then work on improvements and adjust the test thresholds (hopefully down/tighter) as improvements are implemented, right?

Reply:
This is the matter of risk. How likely it is that this test will break things for people running regular PRs on public runners. 

We do not know. We - for now - know that it seems to work on self-hosted. Let's try few times on public runners. I will add label and close/reopem to check.